src/bayesian_classifier.js

```diff
@@ -49,15 +49,15 @@ BayesianClassifier.prototype.train = function(item, category) {
         var v = item[k];
         // Initialize the nested object `data[category][k][item[k]]`
         // with an object of keys that equal 0.
-        if (this.data[category][k] === undefined) {
+        if (!this.data[category][k]) {
```

Comment:
I'd prefer the earlier syntax. What the `!` shortcut really does is

1. Cast the value to a boolean value
2. Compare that value to `false`

This has gotchas: `''` and 0 are values that cast to `false` but still _exist_. So I prefer `===` undefined as a uniform way to test for a lack of a value - otherwise you slip into the habit of using the `!` shortcut often and then introduce a bug where `0` is a legitimate value but is treated as a lack of value.